docs/src/man/backends.md

```diff
@@ -0,0 +1,129 @@
+# Backends and Allocators
+
+The `TensorOperations` package is designed to provide powerful tools for performing tensor computations efficiently.
+In advanced use cases, it can be desirable to squeeze the last drops of performance out of the library, by experimenting with either different micro-optimized implementations of the same operation, or by altering the memory management system.
+Here, we detail how to access these functionalities.
+
+## Backends
+
+### Backend Selection
+
+`TensorOperations` supports multiple backends for tensor contractions, allowing users to choose different implementations based on their specific needs.
+While special care is taken to ensure good defaults, we also provide the flexibility to select a backend manually.
+This can be achieved in a variety of ways:
+
+1. **Global setting**: The default backend can be set globally on a per-type basis, as well as a per-function basis. This is achieved by hooking into the implementation of the default backend selection procedure. In particular, this procedure ends up calling [`select_backend`](@ref)`, which can be overloaded to return a different backend.
+
+2. **Local setting**: Alternatively, the backend can be set locally for a specific call to either [`@tensor`](@ref), [`ncon`](@ref) or the function-based interface. Both `@tensor` and `ncon` accept a keyword argument `backend`, which will locally override the default backend selection mechanism. The result is that the specified backend will be inserted as a final argument to all calls of the primitive tensor operations. This is also how this can be achieved in the function-based interface.
+
+```julia
+using TensorOperations
+mybackend = StridedNative()
+
+# inserting a backend into the @tensor macro
+@tensor backend = mybackend A[i,j] := B[i,k] * C[k,j]
+
+# inserting a backend into the ncon function
+D = ncon([A, B, C], [[1, 2], [2, 3], [3, 1]]; backend=mybackend)
+
+# inserting a backend into the function-based interface
+tensoradd(A, pA, conjA, B, pB, conjB, C, pC, conjC, α, β, mybackend)
```

Comment:
Does this not have too many arguments?

Reply:
`tensoradd(A, pA::Index2Tuple, conjA, B, pB::Index2Tuple, pB::Index2Tuple, conjB, α=1, β=1, [backend])`

Reply:
you also have `pB` twice :grin:

Reply:
Damn, literal copy from the `tensoradd` doc string.

Reply:
interestingly, the docstring on master also has it twice, so I see how this happened